brigade/plugins/tasks/networking/netmiko_file_transfer.py

```diff
@@ -0,0 +1,28 @@
+from brigade.core.task import Result
+
+from netmiko import file_transfer
+
+
+def netmiko_file_transfer(task, source_file, dest_file, **kwargs):
+    """
+    Execute Netmiko file_transfer method
+
+    Arguments:
+        source_file(str): Source file.
+        dest_file(str): Destination file.
+        kwargs (dict, optional): Additional arguments to pass to file_transfer
+
+    Returns:
+        :obj:`brigade.core.task.Result`:
+          * result (``bool``): file exists and MD5 is valid
```

Comment:
I am assuming you are using this to signal the transfer was correct. Shouldn't we return instead bytes written or nothing at all and signal with "failed" if there was some error?

https://github.com/napalm-automation/brigade/blob/develop/brigade/core/task.py#L91

Reply:
It will raise an exception if it tries to transfer and fails. 

'changed' flag will indicate whether the file was transferred or not. It won't transfer the file if the MD5 is already right.

I am using 'result' to indicate the destination file exists and and has the right MD5 (i.e. the file is right).

On your statement what if the file is already right, so the bytes written are 0? This would be confusing to me.

Here is what gets returned from the Netmiko function:

```
{'file_exists': True, 'file_transferred': True, 'file_verified': True}
```

file_exists will always be True (or an exception would be raised)
file_transferred I am mapping to 'changed' in brigade
file_verified indicates whether the MD5 was used for verification or not (since this can be turned off).